sync3/lists.go

```diff
@@ -74,20 +74,32 @@ func (s *InternalRequestLists) SetRoom(r RoomConnMetadata, replacePreviousTimest
 			)
 		}
 
-		// Only bump this room in the room list if the update is of interest to the
-		// client.
-		if replacePreviousTimestamp {
-			r.LastInterestedEventTimestamp = r.LastMessageTimestamp
-		} else {
-			r.LastInterestedEventTimestamp = existing.LastInterestedEventTimestamp
+		// Interpret the timestamp map on r as the changes we should apply atop the
+		// existing timestamps.
```

Comment:
I found it quite confusing to work out the best way to arrange this, and I'm not sure if this is the best way to do it. I think the end result is still a bit smelly.

In the else branch below, we trust that the caller has provided us with sensible timestamps. Doing it this way leaves the door open for us to feed in "accurate" initial timestamps, if we want to address https://github.com/vector-im/element-x-ios/issues/413#issuecomment-1559845357